fix(write): Client.Write fails for []string tags — Serialize falls through to default binary.Write

### Pre-flight Checks
- [x] I have searched existing issues and this is not a duplicate (related but distinct: #59 was the read symmetric, closed by #60; #58 covers server-side STRING)

### Bug Description
`Client.Write(tag, []string{...})` fails locally inside `CIPItem.Serialize` before any payload bytes leave the host. The high-level pipeline already declares support for `[]string` writes (`GoVarToCIPType` registers `[]string`, `write_single` handles slices), but the final serialization step falls through to a `binary.Write` default branch that rejects non-fixed-size slices.

### Steps to Reproduce

**Unit-level (no hardware):**

```go
package gologix

import "testing"

func TestReproSerializeStringSlice(t *testing.T) {
    item := CIPItem{}
    err := item.Serialize([]string{"hello", "world"})
    if err != nil {
        t.Fatalf("Serialize([]string) failed: %v", err)
    }
}
```

**End-to-end against a real controller** — using the standard `gologix_tests` program imported into a CompactLogix L24ER, targeting the existing `ReadStrings` tag (`STRING[10]`):

```go
package gologix_tests

import (
    "testing"

    "github.com/danomagnum/gologix"
)

func TestReproWriteStringArrayHardware(t *testing.T) {
    tcfg := getTestConfig()
    for _, plc := range tcfg.TagReadWriteTests {
        t.Run(plc.PlcAddress, func(t *testing.T) {
            client := gologix.NewClient(plc.PlcAddress)
            if err := client.Connect(); err != nil {
                t.Fatalf("connect: %v", err)
            }
            defer client.Disconnect()

            err := client.Write(
                "program:gologix_tests.ReadStrings",
                []string{"alpha", "bravo", "charlie"},
            )
            if err != nil {
                t.Fatalf("Write([]string) failed: %v", err)
            }
        })
    }
}
```

### Expected Behavior
The three strings get written to consecutive elements of the `STRING[10]` array tag. `Client.Write` returns `nil`.

### Actual Behavior
Session register and forward-open succeed, then `Client.Write` returns an error from inside `CIPItem.Serialize` before any tag write payload is emitted.

### Environment
- Library commit: `082504d` (current `master`)
- Go: 1.21
- Controller: CompactLogix 1769-L24ER, native `STRING` type, `gologix_tests` program imported from `tests/gologix_tests_Program.L5X`
- OS: macOS

### Relevant Logs

Unit-level:
```
=== RUN   TestReproSerializeStringSlice
    Serialize([]string) failed: problem writing default item: binary.Write: some values are not fixed-sized in type []string
--- FAIL: TestReproSerializeStringSlice
```

Hardware:
```
=== RUN   TestReproWriteStringArrayHardware
INFO Session connected
INFO successfully opened connection ConnectionSize=4000
    Write([]string) failed: problem serializing value. problem writing default item: binary.Write: some values are not fixed-sized in type []string
INFO starting disconnection
INFO successfully disconnected from controller
--- FAIL: TestReproWriteStringArrayHardware
```

### Root Cause

`CIPItem.Serialize` (`items.go`) only handles `string` (singular). When `[]string` is passed:

1. `Write` → `write_single`: value is a slice, so `elements = uint16(v.Len())` (`write.go:230-234`).
2. `GoVarToCIPType` returns `CIPTypeSTRING, len(x)` for `[]string` (`types.go:68`), so the IOI footer and element count go out correctly.
3. `reqitems[1].Serialize(value)` is reached with `value = []string{...}`.
4. The `Serialize` switch has `case string` but no `case []string`, so it falls into `default` which calls `binary.Write(item, binary.LittleEndian, []string)`.
5. `encoding/binary` only supports fixed-size types — string slices are not fixed-size, so the call returns the error above.

### Suggested Fix
Add a `case []string:` to the switch in `Serialize` that loops the elements and writes each with the same 4-byte `LEN` + 84-byte fixed-buffer layout already used by `case string:` — the per-element Logix STRING wire shape. Plus unit tests covering empty, single-element, multi-element, and long-string truncation. Happy to send the PR.

### Additional Context
- Scope: client-side `Write` / `WriteMulti` of `[]string` against a Logix controller.
- Out of scope here: server-side STRING handling (#58, planned as a separate PR I already volunteered for).
- Related read-side fix: PR #60 (merged).

--- a/items.go
+++ b/items.go
@@ -231,6 +231,41 @@ func (item *CIPItem) Serialize(strs ...any) error {
 				return fmt.Errorf("problem writing string payload: %v", err)
 			}
 
+		case []string:
+			// Each STRING element on the wire is a fixed 88-byte slot:
+			// 4-byte LEN (uint32) + 84-byte data buffer, the same layout
+			// the controller emits when reading a STRING array.
+			//
+			// Empty or nil slice contributes zero bytes for this argument.
+			// continue keeps the outer variadic loop processing any
+			// remaining arguments instead of returning early.
+			if len(x) == 0 {
+				continue
+			}
+			for _, s := range x {
+				// Clamp the LEN header to the 84-byte payload so the
+				// receiver never sees a header claiming more bytes than
+				// the slot actually carries. copy() below truncates the
+				// payload to 84 bytes; without clamping the header a
+				// >84-byte input would desync the receiver. This is
+				// stricter than the single-string case above, which
+				// writes the raw len(s) into the LEN header even when
+				// the payload gets truncated — out of scope to change
+				// here.
+				strLen := uint32(len(s))
+				if strLen > 84 {
+					strLen = 84
+				}
+				if err := binary.Write(item, binary.LittleEndian, strLen); err != nil {
+					return fmt.Errorf("problem writing string header: %v", err)
+				}
+				b := make([]byte, 84)
+				copy(b, s)
+				if err := binary.Write(item, binary.LittleEndian, b); err != nil {
+					return fmt.Errorf("problem writing string payload: %v", err)
+				}
+			}
+
 		case Serializable:
 			err := binary.Write(item, binary.LittleEndian, x.Bytes())
 			if err != nil {

--- a/items_test.go
+++ b/items_test.go
@@ -1,0 +1,83 @@
+package gologix
+
+import (
+	"encoding/binary"
+	"testing"
+)
+
+// TestSerializeStringSliceEmpty verifies that an empty or nil []string
+// passed to CIPItem.Serialize is treated as a no-op: returns nil and
+// does not write any payload bytes. write_single declares elements=0
+// in the IOI footer for an empty slice, so emitting a partial slot
+// would desync the wire format and the controller would reject the
+// request with a confusing CIP status.
+func TestSerializeStringSliceEmpty(t *testing.T) {
+	cases := []struct {
+		name string
+		in   []string
+	}{
+		{name: "nil", in: nil},
+		{name: "empty", in: []string{}},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			item := CIPItem{}
+			if err := item.Serialize(tc.in); err != nil {
+				t.Fatalf("Serialize(%v) returned error: %v", tc.in, err)
+			}
+			if len(item.Data) != 0 {
+				t.Fatalf("Serialize(%v) wrote %d bytes; want 0", tc.in, len(item.Data))
+			}
+		})
+	}
+}
+
+// TestSerializeStringSliceShape verifies the wire shape for a non-empty
+// []string: each element occupies a fixed 88-byte slot (4-byte LEN +
+// 84-byte data buffer). N elements produce exactly N*88 bytes.
+// long-over-84 also asserts the LEN header is clamped to 84 so the
+// wire header never claims more bytes than the slot actually carries.
+func TestSerializeStringSliceShape(t *testing.T) {
+	cases := []struct {
+		name    string
+		in      []string
+		wantLen uint32
+	}{
+		{name: "single", in: []string{"hello"}, wantLen: 5},
+		{name: "multi", in: []string{"a", "bb", "ccc"}, wantLen: 1},
+		{name: "long-exact-84", in: []string{string(make([]byte, 84))}, wantLen: 84},
+		{name: "long-over-84", in: []string{string(make([]byte, 200))}, wantLen: 84},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			item := CIPItem{}
+			if err := item.Serialize(tc.in); err != nil {
+				t.Fatalf("Serialize(%v) returned error: %v", tc.in, err)
+			}
+			want := len(tc.in) * 88
+			if len(item.Data) != want {
+				t.Fatalf("Serialize(%v) wrote %d bytes; want %d", tc.in, len(item.Data), want)
+			}
+			// LEN header of the first element must match wantLen.
+			gotLen := binary.LittleEndian.Uint32(item.Data[0:4])
+			if gotLen != tc.wantLen {
+				t.Fatalf("first-element LEN header = %d; want %d", gotLen, tc.wantLen)
+			}
+		})
+	}
+}
+
+// TestSerializeStringSliceVariadicContinues verifies that an empty
+// []string sandwiched between other Serialize arguments does NOT
+// short-circuit the variadic loop: arguments before and after the
+// empty slice still get serialized.
+func TestSerializeStringSliceVariadicContinues(t *testing.T) {
+	item := CIPItem{}
+	if err := item.Serialize(uint32(0xDEADBEEF), []string{}, uint32(0xCAFEBABE)); err != nil {
+		t.Fatalf("Serialize(...) returned error: %v", err)
+	}
+	// Two uint32 values = 8 bytes. Empty []string contributes 0.
+	if len(item.Data) != 8 {
+		t.Fatalf("Serialize wrote %d bytes; want 8 (two uint32 with empty []string between)", len(item.Data))
+	}
+}

--- a/tests/write_test.go
+++ b/tests/write_test.go
@@ -1,6 +1,7 @@
 package gologix_tests
 
 import (
+	"fmt"
 	"log"
 	"testing"
 
@@ -376,6 +377,75 @@ func TestWriteUdt(t *testing.T) {
 				t.Errorf("success writing wrong udt: %v", err)
 			}
 
+		})
+	}
+}
+
+// bug: Write/WriteMulti/WriteMap with a []string field fail with
+//   binary.Write: some values are not fixed-sized in type []string
+// because items.Serialize had no case for []string and fell through
+// to binary.Write, which rejects slices of strings.
+// This test writes a []string then reads each element back via the
+// single-tag Read path (which already works) to verify the wire format.
+// The original ReadStrings values are restored element-by-element via
+// the scalar Write path so a regression in the new code can never
+// corrupt the L5X test data.
+func TestWriteStringArray(t *testing.T) {
+	tcs := getTestConfig()
+	for _, tc := range tcs.TagReadWriteTests {
+		t.Run(tc.PlcAddress, func(t *testing.T) {
+			client := gologix.NewClient(tc.PlcAddress)
+			err := client.Connect()
+			if err != nil {
+				t.Error(err)
+				return
+			}
+			defer func() {
+				err := client.Disconnect()
+				if err != nil {
+					t.Errorf("problem disconnecting. %v", err)
+				}
+			}()
+
+			const tag = "program:gologix_tests.ReadStrings"
+
+			// Step 1: backup current values via the working single-tag Read.
+			backup := make([]string, 10)
+			if err := client.Read(tag, backup); err != nil {
+				t.Fatalf("backup read failed: %v", err)
+			}
+
+			// Deferred cleanup (runs after Step 3 below): always restore
+			// originals one-by-one via the scalar write path, which is
+			// independent of the code under test so a regression in the
+			// new code cannot permanently corrupt the L5X dataset.
+			defer func() {
+				for i, v := range backup {
+					idxTag := fmt.Sprintf("%s[%d]", tag, i)
+					if werr := client.Write(idxTag, v); werr != nil {
+						t.Errorf("restore %s failed: %v", idxTag, werr)
+					}
+				}
+			}()
+
+			// Step 2: write a []string with mixed lengths so any per-element
+			// alignment bug surfaces immediately on the read-back.
+			want := []string{"x1", "yy2", "zzz3", "wwww4", "v5", "u66", "t777", "s8888", "r9", "q00"}
+			if err := client.Write(tag, want); err != nil {
+				t.Fatalf("Write([]string) failed: %v", err)
+			}
+
+			// Step 3: read back via the single-tag Read path (already
+			// supports arrays of strings) and verify each element matches.
+			got := make([]string, 10)
+			if err := client.Read(tag, got); err != nil {
+				t.Fatalf("read-back failed: %v", err)
+			}
+			for i := range want {
+				if got[i] != want[i] {
+					t.Errorf("element %d: want %q got %q", i, want[i], got[i])
+				}
+			}
 		})
 	}
 }

